internal/provider/resource/integration_aws_parameter_store.go

```diff
@@ -0,0 +1,501 @@
+package resource
+
+import (
+	"context"
+	"fmt"
+	infisical "terraform-provider-infisical/internal/client"
+
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/listplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+)
+
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ resource.Resource = &IntegrationAWSParameterStoreResource{}
+)
+
+// NewProjectResource is a helper function to simplify the provider implementation.
+func NewIntegrationAwsParameterStoreResource() resource.Resource {
+	return &IntegrationAWSParameterStoreResource{}
+}
+
+// IntegrationAwsParameterStore is the resource implementation.
+type IntegrationAWSParameterStoreResource struct {
+	client *infisical.Client
+}
+
+type AwsParameterStoreMetadataStruct struct {
+	SecretAWSTag []struct {
+		Key   string `json:"key,omitempty"`
+		Value string `json:"value,omitempty"`
+	} `json:"secretAWSTag,omitempty"`
+	ShouldDisableDelete bool `json:"shouldDisableDelete,omitempty"`
+}
```

Comment:
oh I see. we make the metadata property in the request object generic but we add the specific metadata types in the integration resource?